### PR TITLE
fix(gerrit): remove quotes from message filter

### PR DESCRIPTION
**Pull request by kerby82**
Gerrit is not great at providing escape support.
A PR that contains quotes in is message causes Gerrit to return a bad response.
Special characters like quotes are ignored anyway in search so this should not change the expected result of the query.

## Changes

When searching gerrit we remove quotes from the message filter.
Special characters are ignored anyway and escaping support does not work well.

## Context

If the commit message includes quotes the search feature of Gerrit would return a bad response causing
the patchset to not be created.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ X] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ X] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/platform/gerrit/client.ts [hunk 1]
```diff
@@ -219,6 +219,10 @@ class GerritClient {
     return message.substring(0, 0x4000).trim();
   }
 
+  private static removeQuotes(message: string): string {
+    return message.replace(/"/g, '');
```
Comment (viceice) [inline lib/modules/platform/gerrit/client.ts:223]: inline, it's only used once, also use regEx helper function

Comment (kerby82) [reply lib/modules/platform/gerrit/client.ts:223]: Will do! Thanks

**Comment by kerby82** [comment]
👋 Usually how long it takes to have a pull request reviewed? Thanks! 😄

**Comment by viceice** [comment]
please don't force push

File: lib/modules/platform/gerrit/client.ts [hunk 2]
```diff
@@ -12,6 +12,8 @@ import type {
 } from './types';
 import { mapPrStateToGerritFilter } from './utils';
 
+const QUOTES_REGEX = new RegExp('"', 'g');
```
Comment (viceice) [inline lib/modules/platform/gerrit/client.ts:15]:
```suggestion
const QUOTES_REGEX = regEx('"', 'g');
```
use our regex util

**Comment by viceice** [comment]
> 👋 Usually how long it takes to have a pull request reviewed? Thanks! 😄 

use the re-request review button next to the reviewers

**Review by rarkins — DISMISSED** [review]

**Review by viceice — APPROVED** [review]

**Comment by kerby82** [comment]
Build failed with:
```
There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 404 - {'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}
[2024-04-09T10:21:17.448Z] ['verbose'] The error stack is: Error: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 404 - {'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}
```

Seems unrelated to my change but I am unable to retrigger the build

**Pull request merged**